lib/msf/base/sessions/empire.rb

```diff
@@ -0,0 +1,215 @@
+# -*- coding: binary -*-
+require 'msf/ui/console/command_dispatcher.rb'
+require 'msf/base/sessions/command_shell'
+require 'rex/text/table.rb'
+require 'msf/base'
+require 'msf/core/session/interactive.rb'
+
+module Msf
+module Sessions
+
+class EmpireShell < Msf::Sessions::CommandShell
+  include Msf::Session
+  include Rex::Ui::Text::DispatcherShell::CommandDispatcher
+  include Msf::Session::Interactive
+
+  @client_emp = ''
+  @agent_name = ''
+
+  attr_accessor :max_threads
+  attr_accessor :platform
+  attr_accessor :arch
+  attr_accessor :alive
+
+  def desc
+    "Empire Shell #{self.platform}"
+  end
+
+  def interactive?
+    true
+  end
+
+  def self.type
+    "Empire"
+  end
+
+  def framework
+
+  end
+
+  def sid
+
+  end
+
+  def name
+    @agent_name
+  end
+
+  def initialize(emp_object, agent_name)
+    self.platform ||= ""
+    self.max_threads ||= 1
+    self.arch ||= ""
+
+    #Defining instance parameters
+    @client_emp = emp_object
+    @agent_name = agent_name
+  end
+
+  def alive?
+    return self.alive
+  end
+
+
+  #list of available commands
+  def commands
+    {
+      'help'        => 'Show help menu',
+      'show_modules'=> 'List all the Empire post modules available to deploy against the target',
+      'search'      => 'Fetch all the modules resembling to your search term',
+      'show_info'   => 'Displays all the available options and description of the specified module',
+      'use_module'  => 'Attemps to run the specified module against the current agent',
+      'shell'       => 'Run shell commands on target system',
+      'result'      => 'Fetch stored results from the Empire Database',
+      'credentials' => 'Fetch all the saved credentials in the Empire Database',
+      'rename_agent'=> 'Rename the curret agent name for easier interaction',
+      'agentinfo'   => 'Fetch basic information about the agent and host machine'
+    }
+  end
+
+  #
+  #HELP METHODS
+  #------------------
+  #
+  def cmd_help
+    help_table = Rex::Text::Table.new(
+      'Header'        => 'Empire Shell Commands',
+      'HeaderIndent'  => 4,
+      'Prefix'        => '\n',
+      'Postfix'       => '\n',
+      'Coloumns'      => ['Commands','Descriptions'],
+      'Indent'        => 6,
+      'SortIndex'     => -1
+    )
+    commands.each do |command, description|
+      help_table << [command, description]
+    end
+    print(help_table.to_s)
+  end
+
+  #Defining a help method to display command helps.
+  def get_help(usage, example, desc)
+    print_line "Usage : #{usage}"
+    print_line "Example : #{example}"
+    print_line
+    print_line "#{desc}"
+    print_line
+  end
+  #
+  #COMMAND METHODS
+  #----------------------
+  #
+  #Defining show_info command
+  def cmd_show_info(*args)
+    if args.length.zero? || args[0] == '-h' or args[0] == 'help'
```

Comment:
maybe `args.empty?`?
also maybe use either use `&&` and `||` or `and` and `or`

Reply:
Yeah, I actually tried it, but it showed some recursive errors on using `||` or `&&`. I don't yet know if that was only a local problem.